packages/atomic/src/components/commerce/atomic-commerce-pager/e2e/atomic-commerce-pager.e2e.ts

```diff
@@ -0,0 +1,232 @@
+      );
+    });
+
+    test('pager button 2 should not be selected', async ({pager}) => {
+      await expect(pager.numericButton(2)).not.toHaveAttribute(
+        'part',
+        'active-page-button'
+      );
+    });
+
+    test('should not include the first page in the hash', async ({page}) => {
+      await expect(page.url()).not.toContain('#page');
+    });
+
+    test('nextButton should be enabled', async ({pager}) => {
+      await expect(pager.nextButton).toBeEnabled();
+    });
+
+    test('previousButton should be disabled', async ({pager}) => {
+      await expect(pager.previousButton).toBeDisabled();
+    });
+  });
+
+  test.describe('after clicking the pager button', () => {
+    test.beforeEach(async ({pager}) => {
+      await pager.numericButton(3).click();
+    });
+
+    test('pager button 3 should be selected', async ({pager}) => {
+      await expect(pager.numericButton(3)).toHaveAttribute(
+        'part',
+        expect.stringContaining('active-page-button')
+      );
+    });
+
+    test('pager button 1 should not be selected', async ({pager}) => {
+      await expect(pager.numericButton(1)).not.toHaveAttribute(
+        'part',
+        'active-page-button'
+      );
+    });
+
+    test('should include the page in the hash', async ({page}) => {
+      await expect(page.url()).toContain('#page=2');
+    });
+
+    test('nextButton should be enabled', async ({pager}) => {
+      await expect(pager.nextButton).toBeEnabled();
+    });
+    test('previousButton should be enabled', async ({pager}) => {
+      await expect(pager.previousButton).toBeEnabled();
+    });
+  });
+
+  test.describe('after clicking on page 5', () => {
+    test.beforeEach(async ({pager}) => {
+      await pager.numericButton(5).click();
+    });
+
+    test('pager button 5 should not be selected', async ({pager}) => {
+      await expect(pager.numericButton(5)).not.toHaveAttribute(
+        'part',
+        'active-page-button'
+      );
+    });
+  });
+});
+
+test.describe('with a valid page in the hash', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto(
+      'http://localhost:4400/iframe.html?id=atomic-commerce-pager--default&viewMode=story#page=2'
+    );
+  });
+
+  test('pager button 3 should be selected', async ({pager}) => {
+    await expect(pager.numericButton(3)).toHaveAttribute(
+      'part',
+      expect.stringContaining('active-page-button')
+    );
+  });
+
+  test('should be A11y compliant', async ({pager, makeAxeBuilder}) => {
+    await pager.hydrated.waitFor();
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations).toEqual([]);
+  });
+});
+
+test.describe('with an invalid page in the hash', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto(
+      'http://localhost:4400/iframe.html?id=atomic-commerce-pager--default&viewMode=story#page=8k3'
+    );
+  });
+
+  test('pager button 9 should be selected', async ({pager}) => {
```

Comment:
Seems like on first load, the page fragment gets discarded if it has an invalid value... weird!